### ISSUE TITLE
Entry- and Exit-Gateway can be identical

### ISSUE THREAD

**Issue by wiiinnie** [nymtech/nym#4633]
**Describe the issue**
When choosing the same country as entry and exit location you can end up using the same node. So a 2-hop basically turns into a 1-hop vpn

**Expected behaviour**
Expect to have different Entry- / Exit-Gateways selected 

**Stack Traces**
<img width="800" alt="Screenshot 2024-06-07 at 10 33 48" src="https://github.com/nymtech/nym/assets/59703082/b66995ca-fac4-4a90-806c-fe51ec4c6b15">

**Steps to Reproduce**
Steps to reproduce the behaviour, if you're familiar with BDD syntax, please write it in this style:
- Choose country A as entry
- Choose country A as exit
- Connect and disconnect until you end up with the same gateway

*An example:*

**Screenshots**
<img width="800" alt="Screenshot 2024-06-07 at 10 33 48" src="https://github.com/nymtech/nym/assets/59703082/93799bc7-de19-42c3-a21c-2055850d3c02">

**Which area of Nym were you using?**
- NymVPN Version 0.1.4 (20240530.061606) MAC

**Additional context**

**Comment by octol** [comment]
Yep, from the system perspective there is no reason a gateway can't act as both entry and exit for the same client. But I think we want to guard against this happening in the gateway selection process in the vpn client

**Comment by marcdbz** [comment]
Thanks this should be fixed in the newer versions of NymVPN. You can select the same location as entry and exit, but the app will throw an error if the same gateway gets assigned as both entry and exit.
